File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    count = 0
    for index, letter in enumerate(strand_a):
        if strand_a[index] != strand_b[index]:
            count += 1
    return count

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    private int hammingDistance = 0;

    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("strands must be of equal length");
        }

        for (int i = 0; i < leftStrand.length(); i++) {
            if (leftStrand.charAt(i) != rightStrand.charAt(i)) {
                this.hammingDistance++;
            }
        }
    }

    public int getHammingDistance() {
        return hammingDistance;
    }
}
